ionoscloud/resource_nfs_share_test.go

```diff
@@ -0,0 +1,171 @@
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "quota", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "gid", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "uid", "512"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.description", "Client Group 1"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.ip_networks.0", "10.234.50.0/24"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.hosts.0", "10.234.62.123"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.nfs.0.squash", "all-anonymous"),
+				),
+			},
+			{
+				Config: testAccCheckNFSShareConfig_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckNFSShareExists("ionoscloud_nfs_share.example"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "name", "example-share-updated"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "quota", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "gid", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "uid", "1024"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.description", "Client Group 1 Updated"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.ip_networks.0", "10.234.50.0/24"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.hosts.0", "10.234.62.124"),
+					resource.TestCheckResourceAttr("ionoscloud_nfs_share.example", "client_groups.0.nfs.0.squash", "root-anonymous"),
+				),
+			},
+			{
+				Config: testAccDataSourceNFSShareMatchId,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("data.ionoscloud_nfs_share.share_data_example", "name", "ionoscloud_nfs_share.example", "name"),
+					resource.TestCheckResourceAttrPair("data.ionoscloud_nfs_share.share_data_example", "quota", "ionoscloud_nfs_share.example", "quota"),
+					resource.TestCheckResourceAttrPair("data.ionoscloud_nfs_share.share_data_example", "gid", "ionoscloud_nfs_share.example", "gid"),
+					resource.TestCheckResourceAttrPair("data.ionoscloud_nfs_share.share_data_example", "uid", "ionoscloud_nfs_share.example", "uid"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckNFSShareDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(services.SdkBundle).NFSClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ionoscloud_nfs_share" {
+			continue
+		}
+
+		_, resp, err := client.GetNFSShareByID(context.Background(), rs.Primary.Attributes["cluster_id"], rs.Primary.ID, rs.Primary.Attributes["location"])
+		if resp != nil && resp.StatusCode != 404 {
+			return fmt.Errorf("NFS Share still exists: %s", rs.Primary.ID)
```

Comment:
@cristiGuranIonos maybe also provide the cluster ID here.

Reply:
added